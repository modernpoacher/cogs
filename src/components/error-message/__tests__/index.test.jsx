import React from 'react'
import renderer from 'react-test-renderer'

import ErrorMessage from '@modernpoacher/cogs/components/error-message'

jest.useFakeTimers()

describe('@modernpoacher/cogs/components/error-message', () => {
  describe('<ErrorMessage />', () => {
    describe('With required props', () => {
      const component = (
        <ErrorMessage />
      )

      it('renders', () => {
        return expect(renderer.create(component).toJSON())
          .toMatchSnapshot()
      })

      describe('`getClassName`', () => {
        it('is defined', () => {
          return expect(ErrorMessage.prototype.getClassName)
            .toBeDefined()
        })
      })

      describe('`shouldComponentUpdate`', () => {
        it('is defined', () => {
          return expect(ErrorMessage.prototype.shouldComponentUpdate)
            .toBeDefined()
        })
      })
    })

    describe('With additional props', () => {
      it('renders', () => {
        const component = (
          <ErrorMessage
            errorMessage='MOCK ERROR MESSAGE'
          />
        )

        return expect(renderer.create(component).toJSON())
          .toMatchSnapshot()
      })
    })

    describe('`getClassName()`', () => {
      it('returns the classname', () => {
        const component = (
          <ErrorMessage />
        )

        const instance = (
          renderer.create(component)
            .getInstance()
        )

        return expect(instance.getClassName())
          .toBe('error-message')
      })
    })

    describe('`shouldComponentUpdate()`', () => {
      const component = (
        <ErrorMessage
          errorMessage='MOCK ERROR MESSAGE'
        />
      )

      let instance

      beforeEach(() => {
        instance = (
          renderer.create(component)
            .getInstance()
        )
      })

      describe('`props` have changed', () => {
        it('returns true', () => {
          return expect(instance.shouldComponentUpdate({
            errorMessage: 'MOCK CHANGE ERROR MESSAGE'
          }, {
            errorMessage: 'MOCK CHANGE ERROR MESSAGE'
          }))
            .toBe(true)
        })
      })

      describe('`props` have not changed', () => {
        it('returns false', () => {
          return expect(instance.shouldComponentUpdate({ // instance.props
            errorMessage: 'MOCK ERROR MESSAGE'
          }, {
            errorMessage: 'MOCK ERROR MESSAGE'
          }))
            .toBe(false)
        })
      })
    })
  })
})
