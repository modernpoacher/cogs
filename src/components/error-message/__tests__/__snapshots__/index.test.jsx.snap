// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@modernpoacher/cogs/components/error-message <ErrorMessage /> With additional props renders 1`] = `
<span
  className="error-message"
>
  <span
    className="text-content"
  >
    MOCK ERROR MESSAGE
  </span>
</span>
`;

exports[`@modernpoacher/cogs/components/error-message <ErrorMessage /> With required props renders 1`] = `null`;
